### ISSUE TITLE
no-string-based-set-timeout incorrectly flags member methods named setTimeout

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/tslint-microsoft-contrib#439]
```typescript
// Alternately, use @types/jest
const jest = {
    setTimeout: typeof setTimeout,
};

jest.setTimeout(1); // Flagged as an error
```

**Comment by JoshuaKGoldberg** [comment]
Duplicate of #442.
